server/src/main/java/org/elasticsearch/index/fielddata/plain/SortedOrdinalsIndexFieldData.java

```diff
@@ -0,0 +1,103 @@
+import org.elasticsearch.index.fielddata.IndexFieldData.XFieldComparatorSource.Nested;
+import org.elasticsearch.index.fielddata.IndexFieldDataCache;
+import org.elasticsearch.index.fielddata.LeafOrdinalsFieldData;
+import org.elasticsearch.indices.breaker.CircuitBreakerService;
+import org.elasticsearch.script.field.ToScriptField;
+import org.elasticsearch.search.DocValueFormat;
+import org.elasticsearch.search.MultiValueMode;
+import org.elasticsearch.search.aggregations.support.ValuesSourceType;
+import org.elasticsearch.search.sort.BucketedSort;
+import org.elasticsearch.search.sort.SortOrder;
+
+import static org.elasticsearch.index.fielddata.IndexFieldData.XFieldComparatorSource.sortMissingLast;
+
+public class SortedOrdinalsIndexFieldData extends AbstractIndexOrdinalsFieldData {
+
+    public static class Builder implements IndexFieldData.Builder {
+        private final String name;
+        private final ToScriptField<SortedSetDocValues> toScriptField;
+        private final ValuesSourceType valuesSourceType;
+
+        public Builder(String name, ValuesSourceType valuesSourceType, ToScriptField<SortedSetDocValues> toScriptField) {
+            this.name = name;
+            this.toScriptField = toScriptField;
+            this.valuesSourceType = valuesSourceType;
+        }
+
+        @Override
+        public SortedOrdinalsIndexFieldData build(IndexFieldDataCache cache, CircuitBreakerService breakerService) {
+            return new SortedOrdinalsIndexFieldData(cache, name, valuesSourceType, breakerService, toScriptField);
+        }
+    }
+
+    public SortedOrdinalsIndexFieldData(
+        IndexFieldDataCache cache,
+        String fieldName,
+        ValuesSourceType valuesSourceType,
+        CircuitBreakerService breakerService,
+        ToScriptField<SortedSetDocValues> toScriptField
+    ) {
+        super(fieldName, valuesSourceType, cache, breakerService, toScriptField);
+    }
+
+    @Override
+    public SortField sortField(@Nullable Object missingValue, MultiValueMode sortMode, Nested nested, boolean reverse) {
+        SortField sortField = new SortField(getFieldName(), SortField.Type.STRING, reverse);
+        sortField.setMissingValue(
+            sortMissingLast(missingValue) ^ reverse ? SortedSetSortField.STRING_LAST : SortedSetSortField.STRING_FIRST
+        );
+        return sortField;
+    }
+
+    @Override
+    public BucketedSort newBucketedSort(
+        BigArrays bigArrays,
+        Object missingValue,
+        MultiValueMode sortMode,
+        Nested nested,
+        SortOrder sortOrder,
+        DocValueFormat format,
+        int bucketSize,
+        BucketedSort.ExtraData extra
+    ) {
+        throw new IllegalArgumentException("only supported on numeric fields");
+    }
+
+    @Override
+    public LeafOrdinalsFieldData load(LeafReaderContext context) {
+        return new SortedSetBytesLeafFieldData(context.reader(), getFieldName(), toScriptField);
```

Comment:
Might be worth a comment here that `SortedSetBytesLeafFieldData` will happily load SortedDocValues as well, so there's no need for a single-valued implementation.